draft-ietf-httpbis-sfbis.md

```diff
@@ -966,16 +968,23 @@ Given an ASCII string as input_string, return a Date. input_string is modified t
 
 Given an ASCII string as input_string, return a string of Unicode characters. input_string is modified to remove the parsed value.
 
-0. If the first character of input_string is not "%", fail parsing.
-1. Discard the first character of input_string.
-2. Let parsed_string be the result of running Parsing a String ({{parse-string}}) with input_string.
-3. Let byte_array be the result of applying ASCII encoding to input_string.
-4. For each sigil_byte in byte_array which is %25 ("%"):
-   1. Let octet_hex be the two bytes after sigil_byte in byte_string. If there are not two bytes, fail parsing.
-   2. Let octet be the result of decoding octet_hex as hexidecimal, in a case-insensitive fashion.
-   3. Replace sigil_byte and octet_hex in byte_array with octet.
-5. Let unicode_string be the result of decoding byte_array as a UTF-8 string {{UTF8}}. Fail parsing if decoding fails.
-6. Return unicode_string.
+0. If the first two characters of input_string are not  %x25 ("%") followed by DQUOTE, fail parsing.
+1. Discard the first two characters of input_string.
+2. Let byte_array be an empty byte array.
+3. While input_string is not empty:
+   1. Let char be the result of consuming the first character of input_string.
+   2. If char is in the range %x00-1f or %x7f-ff (i.e., it is not in VCHAR or SP), fail parsing.
+   3. If char is %x25 ("%"):
+      1. Let octet_hex be the two bytes after char in input_string. If there are not two bytes, fail parsing.
+      2. Let octet be the result of decoding octet_hex as hexidecimal, in a case-insensitive fashion.
+      3. Append octet to byte_array.
+   2. If char is DQUOTE:
+      1. Let unicode_string be the result of decoding byte_array as a UTF-8 string {{UTF8}}. Fail parsing if decoding fails.
+      2. Return unicode_string.
+   3. Otherwise, if char is not %x25 or DQUOTE:
+      1. Let byte be the result of applying ASCII encoding to char.
```

Comment:
```suggestion
      1. Let byte be the result of applying ASCII decoding to char.
```

Reply:
Converting a character to a byte is encoding it, not decoding it.

Reply:
True. The sentence still reads weird and might be hard to parse for many (I know that "ASCII encoding" of a char actually is a type cast, but how many readers do?)